### PR TITLE
Added context menu options to rearrange Favorites item

### PR DESCRIPTION
**Pull request by BanCrash**
Opening this as a draft since there is still a visual bug that I didn't figure out how to fix it.

I will try to solve this, but since the timeline for the V2 is almost done I suggest to set a timeline also for this, and if I can't solve this issue since it's only visual merge this and open a new issue to fix it in the future.

~~The issue is that when you move the actual selected item, it will stay as selected even if you select another one:~~ partially fixed, now it will work as expected except that moved item won't have the selection pill.

image

**Resolved / Related Issues**
- Related #1069

**Details of Changes**
- Added context menu options to rearrange Favorites item (right click on an item to have to option to move it up/down/top/bottom).

**Current limitations**
- Not available for Home nor Recycle bin. Maybe in a future PR.
- ~~There is a bug when you move the actual selected item, that it will keep it selected even if you switch to another selected item until you select it back again.~~ There is a bug that doesn't show the selection pill when you change the actual selected item.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
image

GIF 30-08-2021 15-49-29

### CODE REVIEW

**Comment by BanCrash** [comment]
I think the bug I've found it's not from Files but from WinUi. I've found this similar issue, though not exactly the same: 
https://github.com/microsoft/microsoft-ui-xaml/issues/2845

And also this that I'm not sure neither if it's the same: https://github.com/microsoft/microsoft-ui-xaml/issues/3015

File: Files/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -2848,4 +2848,16 @@ We use App Center to keep track of app usage, find bugs, and fix crashes. All in
...
+  </data>
+  <data name="SideBarFavoritesMoveOneUp.Text" xml:space="preserve">
+    <value>Move one up</value>
+  </data>
+  <data name="SideBarFavoritesMoveToBottom.Text" xml:space="preserve">
+    <value>Move to bottom</value>
+  </data>
+  <data name="SideBarFavoritesMoveToTop.Text" xml:space="preserve">
+    <value>Move to top</value>
+  </data>
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:2862]:
```suggestion
  <data name="SideBarFavoritesMoveOneDown" xml:space="preserve">
    <value>Move one down</value>
  </data>
  <data name="SideBarFavoritesMoveOneUp" xml:space="preserve">
    <value>Move one up</value>
  </data>
  <data name="SideBarFavoritesMoveToBottom" xml:space="preserve">
    <value>Move to bottom</value>
  </data>
  <data name="SideBarFavoritesMoveToTop" xml:space="preserve">
    <value>Move to top</value>
  </data>
```

File: Files/UserControls/SidebarControl.xaml.cs [hunk 2]
```diff
@@ -930,6 +1026,34 @@ public List<ContextMenuFlyoutItemViewModel> GetLocationItemMenuItems()
...
+                    Glyph = "\uE70D",
+                    Command = new RelayCommand(() => MoveItemDown_Click(null, null)),
+                    ShowItem = ShowMoveItemDown
+                },
+                new ContextMenuFlyoutItemViewModel()
+                {
+                    Text = "SideBarFavoritesMoveToBottom/Text".GetLocalized(),
+                    Glyph = "\uE118",
+                    Command = new RelayCommand(() => MoveItemToBottom_Click(null, null)),
+                    ShowItem = ShowMoveItemDown
```
Comment (yaichenbaum) [inline Files/UserControls/SidebarControl.xaml.cs:1054]:
```suggestion
                    Text = "SideBarFavoritesMoveToTop".GetLocalized(),
                    Glyph = "\uE11C",
                    Command = new RelayCommand(() => MoveItemToTop_Click(null, null)),
                    ShowItem = ShowMoveItemUp
                },
                new ContextMenuFlyoutItemViewModel()
                {
                    Text = "SideBarFavoritesMoveOneUp".GetLocalized(),
                    Glyph = "\uE70E",
                    Command = new RelayCommand(() => MoveItemUp_Click(null, null)),
                    ShowItem = ShowMoveItemUp
                },
                new ContextMenuFlyoutItemViewModel()
                {
                    Text = "SideBarFavoritesMoveOneDown".GetLocalized(),
                    Glyph = "\uE70D",
                    Command = new RelayCommand(() => MoveItemDown_Click(null, null)),
                    ShowItem = ShowMoveItemDown
                },
                new ContextMenuFlyoutItemViewModel()
                {
                    Text = "SideBarFavoritesMoveToBottom".GetLocalized(),
                    Glyph = "\uE118",
                    Command = new RelayCommand(() => MoveItemToBottom_Click(null, null)),
                    ShowItem = ShowMoveItemDown
```

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

**Comment by R3voA3** [comment]
While I like the idea of being able to change the order of appearance, wouldn't it be more intuitive to just be able to drag and drop the items?

**Comment by BanCrash** [comment]
> While I like the idea of being able to change the order of appearance, wouldn't it be more intuitive to just be able to drag and drop the items? With the context menu, rearranging many items can become tedious.

My idea would be to have both ways and everyone choose whatever option they want (or even sometimes one and sometimes the other). But currently there is a but on WinUI that doesn't allow to have the drag and drop functionality. When that bug is fixed we can have also that way.

**Comment by R3voA3** [comment]
> > While I like the idea of being able to change the order of appearance, wouldn't it be more intuitive to just be able to drag and drop the items? With the context menu, rearranging many items can become tedious.
> 
> My idea would be to have both ways and everyone choose whatever option they want (or even sometimes one and sometimes the other). But currently there is a bug on WinUI that doesn't allow to have the drag and drop functionality. When that bug is fixed we can have also that way.

Sounds great!

**Comment by yaichenbaum** [comment]
> While I like the idea of being able to change the order of appearance, wouldn't it be more intuitive to just be able to drag and drop the items? With the context menu, rearranging many items can become tedious.

As @BanCrash mentioned, drag and drop would be ideal, but until the WinUI issue is resolved, we're implementing this as a workaround.

**Review by gave92 — APPROVED** [review]
Nice!

**Review by yaichenbaum — APPROVED** [review]

**Pull request merged**